src/ide/svg.ts

```diff
@@ -0,0 +1,179 @@
+export function createSvg(width: string, height: string): SVGSVGElement {
+	let svg: SVGSVGElement = document.createElementNS(
+		"http://www.w3.org/2000/svg",
+		"svg"
+	);
+	svg.setAttribute("width", width);
+	svg.setAttribute("height", height);
+	return svg;
+}
+
+function createSvgElement(type, properties) {
+	let el = document.createElementNS("http://www.w3.org/2000/svg", type);
+	for (let p in properties) {
+		if (properties.hasOwnProperty(p)) el.setAttribute(p, properties[p]);
+	}
+	return el;
+}
+
+export class SVGDrawingContext {
+	public readonly svg: SVGSVGElement;
+	#classname: string;
+	#style: string;
+	#path?: string;
```

Comment:
Concerning ES private class member syntax: please don't use it. It's only a very recent feature (Firefox has it since July) and Safari doesn't yet support private class methods.

In exchange, just use TypeScript's [visibility modifiers](https://www.typescriptlang.org/docs/handbook/2/classes.html#member-visibility). They ensure the correct visibility at compile time which is good enough.